tempest/api/workloadmgr/security_groups/test_security_group_default.py

```diff
@@ -0,0 +1,322 @@
+        LOG.debug("sec group + rules: {}".format(secgrp_list_with_rules))
+        return secgrp_list_with_rules
+
+    """ Method to create Security groups with distinct rules """
+
+    def _create_sec_groups_rule(self, security_group_names, data_sec_group_and_rules):
+        LOG.debug("Create security group and rule")
+        created_security_groups = []
+        for each in range(1, len(data_sec_group_and_rules) + 1):
+            sgid = self.create_security_group(
+                name=security_group_names + format(each),
+                description="security group with distinct rules",
+                secgrp_cleanup=True,
+            )
+            created_security_groups.append(sgid)
+            # Delete default security group rules
+            self.delete_default_rules(sgid)
+        for t in range(0, len(created_security_groups)):
+            LOG.debug(
+                "Creating rule with other details "
+            )
+            secgrp = created_security_groups[t]
+            for each in data_sec_group_and_rules[t]:
+                self.add_security_group_rule(
+                    parent_grp_id=secgrp,
+                    ip_proto=each["protocol"],
+                    from_prt=each["port_range_min"],
+                    to_prt=each["port_range_max"],
+                )
+        return created_security_groups
+
+    def _security_group_and_rules_verification(self, security_group_names, data_sec_group_and_rules, rules_count):
+        for t in range(0, len(data_sec_group_and_rules)):
+            restored_security_group = security_group_names + str(t + 1)
+            sgid = self.get_security_group_id_by_name(restored_security_group)
+            # restored security groups to be cleaned after the execution
+            self.restored_secgroup_ids.append(sgid)
+            rule_list = self.list_secgroup_rules_for_secgroupid(sgid)
+            count = 0
+
+            for each in data_sec_group_and_rules[t]:
+                for rule in rule_list:
+                    result = self.verifySecurityGroupRules(rule["id"], sgid, each)
+                    if result == True:
+                        count += 1
+                LOG.debug(
+                    "Here is the count: {} for restored rules per security group {}".format(
+                        count, each
+                    )
+                )
+
+            if count == rules_count * 2:
+                reporting.add_test_step(
+                    "Security group rules verification successful for restored secgroup {}".format(
+                        restored_security_group
+                    ),
+                    tvaultconf.PASS,
+                )
+            else:
+                reporting.add_test_step(
+                    "Security group rules verification failed for restored secgroup {}".format(
+                        restored_security_group
+                    ),
+                    tvaultconf.FAIL,
```

Comment:
test script status also need to be marked as FAIL explicitly

Reply:
done